### ISSUE TITLE
barrel preset, adding extension for support ESM modules

### ISSUE THREAD

**Issue by zanza00** [mmkal/eslint-plugin-codegen#12]
One of the requirements of ESM modules is that every import need an extension ([see here](https://nodejs.org/api/esm.html#import-specifiers)).

Since it depends on the build toolchain on exactly what extension to use, it can be anyone of the followings `js`, `mjs`, `cjs`, `ts`, `mts` and `cts`.

I suggest adding a specific option called `extension` that is appended to the import, using the example in the readme it will be modified as follows

```ts
// codegen:start {preset: barrel, include: some/path/*.ts, exclude: some/path/*util.ts, extension: mjs}
export * from './some/path/module-a.mjs'
export * from './some/path/module-b.mjs'
export * from './some/path/module-c.mjs'
// codegen:end
```

I glanced the code and I think I can create a PR

**Comment by mmkal** [comment]
Makes sense to me - although I think `extension: true` would be better, that would allow for mixed  `.js`, `.ts`, `.mjs`, `.tsx` etc. A PR along those lines would be very welcome!

**Comment by zanza00** [comment]
I think that we should allow for an override of some sort, in my project I am using `.ts` files but I am importing `.js` files. This is the default behavior of using Typescript 5 and using `"module": "NodeNext"` in the `tsconfig.json`.

We could use `extension: boolean | string` for a blanket override or use a map of some sort like `extension: true, map: { ts: 'js', js: 'cjs'}`

**Comment by mmkal** [comment]
Ah, right. Makes sense. I can see the use for `boolean | string | Record<string, string>`. Would you like to create a PR for that?

**Comment by zanza00** [comment]
I created a draft PR with a test, I will update it shortly with the record. If you have any early feedback feel free to comment it.

**Comment by zanza00** [comment]
opened PR #13 

I am not super happy about the implementation, any feedback is welcome

**Comment by mmkal** [comment]
released in v0.18.0
